modules/phase_field/include/actions/GrainGrowthNonlinPrecondAction.h

```diff
@@ -0,0 +1,40 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "Action.h"
+
+#include "libmesh/fe_type.h"
+
+// Forward declaration
+
+class GrainGrowthNonlinPrecondAction : public Action
```

Comment:
This looks similar enough to `GrainGrowthAction` that some refactoring would be in order.

Reply:
I addressed this comment and refactored GrainGrowthAction and now this action inherits from it.